src/docker/composebuilder/service_builder.go

```diff
@@ -153,11 +153,16 @@ func (d *ServiceComposeBuilder) getDockerEnvVars() map[string]string {
 			result[variable] = value
 		}
 	}
-	envVars := d.ServiceConfig.Local.Environment
-	util.Merge(result, envVars)
+	serviceEnvVars := d.ServiceConfig.Local.Environment
```

Comment:
I think `serviceEnvVars` and `envVars` are unnecessary variables. We can put the reference inline the `util.Merge`